pig/src/main/java/org/jboss/pnc/bacon/pig/impl/documents/sharedcontent/BrewSearcher.java

```diff
@@ -37,27 +41,59 @@
 public class BrewSearcher {
     private static final Logger log = LoggerFactory.getLogger(BrewSearcher.class);
 
-    public static void fillBrewData(SharedContentReportRow row) {
-        log.debug("Asking for {}\n", row.toGapv());
-        List<KojiBuild> builds = getBuilds(row);
-        builds.forEach(build -> {
-            int buildId = build.getBuildInfo().getId();
-            log.debug("Got build id {} for artifact {}", buildId, row.toGapv());
-
-            if (row.getBuildId() != null) {
-                row.setBuildId(row.getBuildId() + ", " + buildId);
-            } else {
-                row.setBuildId(String.valueOf(buildId));
+    public static void fillBrewData(List<SharedContentReportRow> rows) {
+        List<String> pathsForRows = rows.stream()
+                .map(row -> row.getFilePath().toAbsolutePath().toString())
+                .collect(Collectors.toList());
+
+        Map<GAV, SharedContentReportRow> rowsByGav = rows.stream()
+                .collect(Collectors.toMap(SharedContentReportRow::getGav, Function.identity()));
+
+        List<KojiBuild> builds;
+        int attempts = 3;
```

Comment:
So, it is still failing sometimes?

Reply:
I don't know, we can try to remove it, I left it from the previous version

Reply:
I would like to know if it's fixed. Something should be done upstream to fix this, although I am not sure what at this point.